src/Microsoft.ML/Runtime/EntryPoints/ImportTextData.cs

```diff
@@ -27,13 +27,25 @@ public sealed class Input
             public string CustomSchema = null;
         }
 
+        [TlcModule.EntryPointKind(typeof(ILearningPipelineLoader))]
+        public sealed class LoaderInput
+        {
+            [Argument(ArgumentType.Required, ShortName = "data", HelpText = "Location of the input file", SortOrder = 1)]
+            public IFileHandle InputFile;
+
+            [Argument(ArgumentType.Required, ShortName = "args", HelpText = "Arguments", SortOrder = 2)]
+            public TextLoader.Arguments Arguments = new TextLoader.Arguments();
+        }
+
         public sealed class Output
         {
             [TlcModule.Output(Desc = "The resulting data view", SortOrder = 1)]
             public IDataView Data;
         }
 
-        [TlcModule.EntryPoint(Name = "Data.TextLoader", Desc = "Import a dataset from a text file")]
+#pragma warning disable 0618
+        [Obsolete("Use TextLoader instead.")]
```

Comment:
>Obsolete [](start = 9, length = 8)

Is this the right place to have this attribute?
If a user has code that calls Experiment.Add(new TextLoader() {...}) with the old API, won't that code break, since TextLoader will now have different fields?

Reply:
Yes, that is the whole point. We want to discourage people from using this API.

---
In reply to: [188734634](https://github.com/dotnet/machinelearning/pull/142#discussion_r188734634) [](ancestors = 188734634)